pkgs/development/python-modules/mesa/default.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, isPy27
+, cookiecutter, networkx , pandas, tornado, tqdm }:
+
+buildPythonPackage rec {
+  pname = "mesa";
+  version = "0.8.6";
+
+  src = fetchFromGitHub {
+    repo = "mesa";
+    owner = "projectmesa";
+    rev = "v${version}";
+    sha256 = "0d8c636zhswxd91ldlmdxxlyym2fj3bk1iqmpc1jp3hg7vvc7w03";
+  };
+
+  # According to their docs, this library is for Python 3+.
+  disabled = isPy27;
+
+  propagatedBuildInputs = [ cookiecutter networkx pandas tornado tqdm ];
+
```

Comment:
Please add tests
```suggestion
checkInputs = [ pytest ];
checkPhase = ''
  pytest tests
'';
```

Reply:
Yes, of course. I just did, but as of now, one of the tests fails. I think it has something to do with it being run by Nix because it tests whether a directory is correctly created by some procedure; for that it uses 'click.testing'’s `runner.isolated_filesystem` (which is used to run code in a new temporary folder). However, the test does not find the created directory afterwards. Perhaps you can give me a hint?

Reply:
do you mind pasting the results?

Reply:
actually, i did it for you:

```
checkInputs = [ pytest ];

  checkPhase = ''
    pytest tests -k 'not scaffold'
  '';
```

with pytest, you can use the `-k` option to specify a predicate on test case names, or you can use `--ignore=path/to/test/file.py` to ignore entire files/directories.

Reply:
Ah, OK, thanks a lot. I did not know that simply disabling the problematic test was a viable option.

Reply:
it is if it's something out of our control. Since nixpkgs builds in a sandbox environment, a lot of system changing things may fail. Which is fine. 

At the very least, we should be trying to enable as many tests as we can.